cluster-up/cluster/kind-k8s-sriov-1.17.0/certcreator/certlib/selfsign.go

```diff
@@ -0,0 +1,115 @@
+package certlib
+
+import (
+	"bytes"
+	cryptorand "crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"math/big"
+	"math/rand"
+	"time"
+)
+
+type SelfSignedCertificate struct {
+	DNSNames    []string
+	CommonName  string
+	Certificate *bytes.Buffer
+	PrivateKey  *bytes.Buffer
+}
+
+func (s *SelfSignedCertificate) Generate() error {
+	var caPEM *bytes.Buffer
+
+	randomSource := rand.New(rand.NewSource(time.Now().Unix()))
+	caCertificateConfig := &x509.Certificate{
+		SerialNumber: big.NewInt(randomSource.Int63()),
+		Subject: pkix.Name{
+			Organization: []string{"kubvirt.io"},
+		},
+		NotBefore:             time.Now(),
+		NotAfter:              time.Now().AddDate(1, 0, 0),
+		IsCA:                  true,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth, x509.ExtKeyUsageServerAuth},
+		KeyUsage:              x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
+		BasicConstraintsValid: true,
+	}
+
+	caPrivateKey, err := rsa.GenerateKey(cryptorand.Reader, 4096)
+	if err != nil {
+		return fmt.Errorf("faild to generate CA private key: %v", err)
```

Comment:
/g was because faild is all over
(also wrote there its all over please)

Reply:
Done